FunctionalArray.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  FunctionalArray.swift
+//  SwiftTrainingExercise
+//
+//  Created by Ahmed Ibrahim on 20/06/2023.
+//
+
+import Foundation
+
+class FunctionalArray<Element> where Element: Equatable & Comparable & Hashable {
+  
+  var elementsArray: [Element] = [] // how to set this as private only for subclass and superclass so its not accessable ?
+  private(set) var maxSize = Int.max;
```

Comment:
why do you have a default value ?

Reply:
mostly just not to use the optional and check for nil, and i thought it would be more logical to have as max since if we did not specify a size for it, we can add as much elements .  (assuming its max capacity is int.max)